pym/bob/input.py

```diff
@@ -2373,6 +2373,10 @@ def __parse(self):
 
         # finally parse recipes
         if not os.path.isdir("recipes"):
+            self.__cache.close()
```

Comment:
What about moving this test to Line 2196?
```
    def __init__(self):
        if not os.path.isdir("recipes"):
            raise ParseError("No recipes directory found.")
```

to be safe(er) on futher addings?